### PR TITLE
Category graphql/improve api flow

### PR DESCRIPTION
**Pull request by sundersc**
Improve the add and update api workflow by reducing the number of steps required to setup or update an AppSync GraphQL API resource.

#### Description of changes

- add api - Improve from current 10 minimum steps required to setup graphql api to 4 steps.
- update api - display general information about the graphql api. Improve the workflow to reduce the number of steps required to update a resource.
- add default templates for V2.
- auto apply auth modes to AppSync API based on schema.

#### Description of how you validated changes
- Manual test
- yarn test passes
- e2e tests in local

#### Checklist
- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts [hunk 1]
```diff
@@ -330,111 +479,122 @@ export const updateWalkthrough = async (context): Promise<UpdateApiRequest> => {
...
+async function askResolverConflictQuestion(context, resolverConfig, modelTypes?) {
+  let resolverConfigResponse: any = {};
 
-      if (conflictResolutionStrategy === 'LAMBDA') {
-        const { newFunction, lambdaFunctionName } = await askSyncFunctionQuestion(context);
-        syncConfig.LambdaConflictHandler = {
-          name: lambdaFunctionName,
-          new: newFunction,
-        };
+  if (await context.prompt.confirm('Enable conflict detection?', !resolverConfig?.project)) {
```
Comment (danielleadams) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts:535]: Should we be using `amplify-prompts` here?

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts [hunk 2]
```diff
@@ -46,6 +47,63 @@ const authProviderChoices = [
...
+    name: 'Optimistic Concurrency',
+    value: 'OPTIMISTIC_CONCURRENCY',
+  },
+  {
+    name: 'Custom Lambda',
+    value: 'LAMBDA',
+  },
+  {
+    name: 'Learn More',
+    value: 'Learn More',
```
Comment (danielleadams) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts:65]: Should this have a key value that's all caps like the other options?

File: packages/amplify-category-api/src/provider-utils/awscloudformation/utils/edit-schema-flow.ts [hunk 3]
```diff
@@ -8,7 +8,7 @@ export const editSchemaFlow = async (context: any, apiName: string) => {
     type: 'confirm',
     name: 'editNow',
     message: 'Do you want to edit the schema now?',
-    default: false,
+    default: true,
```
Comment (danielleadams) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/utils/edit-schema-flow.ts:11]: was this supposed to be switched?

File: packages/amplify-category-api/src/index.ts [hunk 4]
```diff
@@ -217,3 +219,19 @@ export async function handleAmplifyEvent(context, args) {
...
+export async function addGraphQLAuthorizationMode(context, args) {
+  const { authType, printLeadText, authSettings } = args;
+  const apiName = getAppSyncResourceName(context.amplify.getProjectMeta());
+  if (!apiName) {
+    return;
+  }
+
+  const authConfig = getAppSyncAuthConfig(context.amplify.getProjectMeta());
+  const addAuthConfig = await askAuthQuestions(authType, context, printLeadText, authSettings);
+  authConfig.additionalAuthenticationProviders.push(addAuthConfig);
```
[the remark this answers is not in the document]
Comment (danielleadams) [reply packages/amplify-category-api/src/index.ts:232]: I see where this is used now, can the function name be like addAdditionalGraphQLAuthorizationMode so it's clear what this is for.

**Review by danielleadams — COMMENTED** [review]
Just a few comments, but functionality looks there. I'm going to test locally again too (already did at bug bash, so I know what it should look like).

File: packages/amplify-category-api/src/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (sundersc) [reply packages/amplify-category-api/src/index.ts:232]: Default Auth mode is an object whereas additionalAuth is an array. So 'add'/'remove' refers to the additional auth array.

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts (hunk 2 above)
Comment (sundersc) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts:65]: This is an existing piece of code, we can clean it up with a different PR.

File: packages/amplify-category-api/src/provider-utils/awscloudformation/utils/edit-schema-flow.ts (hunk 3 above)
Comment (sundersc) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/utils/edit-schema-flow.ts:11]: Yes, this is intentional. We are changing the default option to 'Y' for this prompt.

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts [hunk 5]
```diff
@@ -46,6 +47,63 @@ const authProviderChoices = [
...
+  {
+    name: 'Objects with fine-grained access control (e.g., a project management app with owner-based authorization)',
+    value: 'single-object-auth-schema-v2.graphql',
+  },
+  {
+    name: 'Blank Schema',
+    value: 'blank-schema.graphql',
+  },
+];
+
```
[the remark this answers is not in the document]
Comment (sundersc) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts:106]: That's a good point. For now, I had to go with how the existing prompts are configured in this service walkthrough. I can create a work item to address this in the future.

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts (hunk 1 above)
Comment (sundersc) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts:535]: These are existing prompts, I haven't planned to migrate these to use amplify-prompts in this PR. We need to have a separate work item.

File: packages/amplify-category-api/src/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (danielleadams) [reply packages/amplify-category-api/src/index.ts:232]: Got it... okay, that makes sense.

File: packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (danielleadams) [reply packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-walkthrough.ts:106]: As long as we have work items to address this, that works for me.

**Review by danielleadams — APPROVED** [review]

**Pull request merged**